compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -649,29 +649,335 @@ abstract class GenJSCode extends plugins.PluginComponent
         constructorTrees: List[DefDef]): js.Tree = {
       implicit val pos = classSym.pos
 
-      val (primaryCtorTree :: Nil, secondaryCtorTrees) =
-        constructorTrees.partition(_.symbol.isPrimaryConstructor)
-
       // Implementation restriction
-      val sym = primaryCtorTree.symbol
+      val syms = constructorTrees.map(_.symbol)
       val hasBadParam = enteringPhase(currentRun.uncurryPhase) {
-        sym.paramss.flatten.exists(p => p.hasDefault || isRepeated(p))
+        syms.exists(_.paramss.flatten.exists(p => p.hasDefault))
       }
       if (hasBadParam) {
         reporter.error(pos,
             "Implementation restriction: the constructor of a " +
-            "Scala.js-defined JS classes cannot have default parameters nor " +
-            "repeated parameters.")
+            "Scala.js-defined JS classes cannot have default parameters.")
```

Comment:
The following assertion fails: https://github.com/scala-js/scala-js/pull/1982/files#diff-356ccceef3871dbfcfa92b93f89a23c1R627. In these cases the issue seems to come from trying to access the the class module with:
```scala
val trgSym = {
  if (sym.isClassConstructor) sym.owner.companionModule.moduleClass
  else sym.owner
}
```
where `sym.isClassConstructor` is `true` and `sym.owner.companionModule` is `<none>`. 

I tried a few different ways to access the module but wasn't able. Therefore I imagine that this symbol has something broken and we might need to do something more involved to have access to the default parameters if they still actually exist.

Could the prejsinterop be messing with these modules?